test/sharness/t0020-init.sh

```diff
@@ -58,11 +58,13 @@ test_expect_success "ipfs config succeeds" '
 	test_cmp expected_config actual_config
 '
 
+test_check_peerid() {
```

Comment:
Yeah, this could be moved to the library and used in t0062-daemon-api.sh too.
I don't understand why you say that it could replace the changes you link to though.